packages/middleware-recursion-detection/src/index.ts

```diff
@@ -0,0 +1,64 @@
+import { HttpRequest } from "@aws-sdk/protocol-http";
+import {
+  AbsoluteLocation,
+  BuildHandler,
+  BuildHandlerArguments,
+  BuildHandlerOptions,
+  BuildHandlerOutput,
+  BuildMiddleware,
+  MetadataBearer,
+  Pluggable,
+} from "@aws-sdk/types";
+
+const TRACE_ID_HEADER_NAME = "X-Amzn-Trace-Id";
+const ENV_LAMBDA_FUNCTION_NAME = "AWS_LAMBDA_FUNCTION_NAME";
+const ENV_TRACE_ID = "_X_AMZN_TRACE_ID";
+
+interface PreviouslyResolved {
+  runtime: string;
+}
+
+/**
+ * Inject to trace ID to request header to detect recursion invocation in Lambda.
+ * @internal
+ */
+export function recursionDetectionMiddleware(options: PreviouslyResolved): BuildMiddleware<any, any> {
+  return <Output extends MetadataBearer>(next: BuildHandler<any, Output>): BuildHandler<any, Output> =>
```

Comment:
Convert to arrow function
```suggestion
const recursionDetectionMiddleware = (options: PreviouslyResolved): BuildMiddleware<any, any>  => 
  (next: BuildHandler<any, Output>): BuildHandler<any, Output> =>
```